### PR TITLE
Big Compute docs clarification blitz

### PR DESCRIPTION
**Pull request by LauraLangdon**
WIP update to SCN docs, more details to come

Edit: this is now semi-complete, with more updates planned for the pages on namespaces, configuring storage for cloud deployment, FQFN, custom templates, and custom libraries.

### CODE REVIEW

**Comment by flaki** [comment]
Preview branch deploy for this PR: https://suborbital.github.io/docs/laura/update-scn-docs/
(it's always `https://suborbital.github.io/docs/` + the branch name + `/` )

File: website/docs/compute/get-started.md
Comment (flaki) [inline website/docs/compute/get-started.md:76]: It would be important to highlight that they need to run these commands (particularly `docker-compose down`) in the same directory, that is, the directory where the `docker-compose.yaml` lives.

We might also want to mention suborbital/subo#344 - that is, that some systems only have `docker compose` and `docker-compose` will fail

File: website/docs/compute/get-started.md
Comment (flaki) [inline website/docs/compute/get-started.md:73]: Note that on some systems this will fail ( suborbital/subo#345 ) before we even told people about the `--proxy-port` option.

File: website/docs/compute/get-started.md
Comment (flaki) [inline website/docs/compute/get-started.md:108]: Can we keep the function name consistent across the docs please?
On the function editor image above we use `hello`, so let's stick with that?

File: website/docs/compute/get-started.md
Comment (flaki) [inline website/docs/compute/get-started.md:140]:
```suggestion
http://local.suborbital.network/?builder=http://local.suborbital.network:8082&token=ogbWUbzCZaYLga3GggP0cxht&ident=com.suborbital.acmeco&namespace=default&fn=hello
```

**Review by flaki — CHANGES_REQUESTED** [review]
@LauraLangdon you will need to update the Node.js version [.github/workflows/documentation.yml](https://github.com/suborbital/docs/blob/580ddad6d8ed1eb6c93f5a07291256479abb4c19/.github/workflows/documentation.yml#L16) from `14` to `16` for the automatic build deploys to work!

File: website/docs/compute/get-started.md
Comment (LauraLangdon) [reply website/docs/compute/get-started.md:76]: I've added the cautionary points about about running in the correct directory, but we'd need to pair for the docker part, and I'm thinking it needn't hold up merging this for now. I'll create an issue for it!

File: website/docs/compute/get-started.md
Comment (LauraLangdon) [reply website/docs/compute/get-started.md:73]: I'll create an issue to document this after this one merges.

File: .github/workflows/documentation.yml
Comment (callahad) [inline .github/workflows/documentation.yml:18]:
```suggestion
```

Do we still need this, now that we're specifying Node 16?

File: website/docs/compute/customizing-functions/fully-qualified-function-names.md
Comment (callahad) [inline website/docs/compute/customizing-functions/fully-qualified-function-names.md:19]: Is escaping the closing paren necessary?

File: website/docs/compute/customizing-functions/custom-function-templates.md
Comment (callahad) [inline website/docs/compute/customizing-functions/custom-function-templates.md:16]: Do we have a public tracking bug we can link here?

File: website/docs/compute/customizing-functions/fully-qualified-function-names.md
Comment (callahad) [inline website/docs/compute/customizing-functions/fully-qualified-function-names.md:37]: Should this be `com.example` to match earlier prose?

File: website/docs/compute/get-started.md
Comment (callahad) [inline website/docs/compute/get-started.md:81]: In light of suborbital/subo#344, maybe we should replace references to `docker-compose` with `docker compose`? (Could be handled in a follow-up PR, rather than prolonging this one... but if so, we should make sure it's tracked somewhere.

File: website/docs/compute/customizing-functions/code-editor.md
Comment (callahad) [inline website/docs/compute/customizing-functions/code-editor.md:23]: Weird line breaking here. Do we care?

**Review by callahad — CHANGES_REQUESTED** [review]
In addition to the inline comments, I'd request changes:

- The `website/docs/assets/editor-screen.png` image still seems *huge* (over 300k!)... I think we should crop it further, reduce its resolution, and send it through a compressor.

I'd also like to call out but leave to your discretion:

- It seems like many of the prose changes switch from one sentence per line to hard-wrapping at ~80 characters. Others do not hard wrap. Others seem to break lines at arbitrary locations. Do we care if this is consistent?

**Comment by ospencer** [comment]
@callahad after this is merged I plan to do https://github.com/suborbital/docs/issues/162 to ensure we've got consistent formatting.

File: website/docs/compute/compute.md
Comment (ospencer) [inline website/docs/compute/compute.md:6]: A bit of a nit, but it looks like this (and the header removal) are the only things that changed in this file. Ideally, other changes like formatting are left out and done in separate PRs. It makes it much easier to track what's actually changing!

File: website/docs/compute/customizing-functions/code-editor.md
Comment (ospencer) [inline website/docs/compute/customizing-functions/code-editor.md:35]:
```suggestion
* This is the URL of your public builder service
```

File: website/docs/compute/customizing-functions/custom-function-templates.md
Comment (ospencer) [reply website/docs/compute/customizing-functions/custom-function-templates.md:16]: Nothing public. Private tracking issue: https://github.com/suborbital/scn/issues/556

File: website/docs/compute/deployment/cloud-deployment/configure-webhooks.md
Comment (ospencer) [inline website/docs/compute/deployment/cloud-deployment/configure-webhooks.md:46]: I know we mentioned that this seems weird being a list with only one thing in it, but now that I see this I feel like it's weirder. `Hook point` in particular feels very off.

File: website/docs/compute/get-started.md
Comment (ospencer) [inline website/docs/compute/get-started.md:86]: This example really shows how confusing the repl is.

File: website/docs/compute/using-api.md
Comment (ospencer) [inline website/docs/compute/using-api.md:154]:
```suggestion
    adminUri: "https://acme.co:8081",
    execUri: "https://acme.co:8080",
    builderUri: "https://acme.co/builder",
```

File: website/docs/compute/using-api.md
Comment (ospencer) [inline website/docs/compute/using-api.md:177]:
```suggestion
async function listAvailableFunctions() {
    const functionList = await suborbital.admin.getFunctions({
        userId: "1234",
        namespace: "default",
    });

    console.log("Functions:", functions);
}
```

File: website/docs/compute/using-api.md
Comment (ospencer) [inline website/docs/compute/using-api.md:214]:
```suggestion
async function runFunction() {
    const result = await suborbital.exec.run({
        environment: "com.acmeco",
        userId: "1234",
        namespace: "default",
        fnName: "foo",
        version: "v1.0.0",
    });

    console.log("Function output:", result);
}
```

File: website/docs/compute/using-api.md
Comment (ospencer) [inline website/docs/compute/using-api.md:248]:
```suggestion
    "functions": [
        {
            "name": "foo",
            "namespace": "...",
            "lang": "...",
            "version": "...",
            "draftVersion": "...",
            "apiVersion": "...",
            "fqfn": "...",
            "fqfnURI": "...",
        }
    ]
```

File: website/docs/compute/using-api.md
Comment (ospencer) [inline website/docs/compute/using-api.md:266]:
```suggestion
    "results": [
        {
            "uuid": "...",
            "timestamp": "...",
            "response": "..."
        }
    ]
```

File: website/docs/compute/using-api.md
Comment (ospencer) [inline website/docs/compute/using-api.md:285]:
```suggestion
    "errors": [
        {
            "uuid": "...",
            "timestamp": "...",
            "code": 400,
            "message": "..."
        }
    ]
```

File: website/docs/compute/using-api.md
Comment (ospencer) [inline website/docs/compute/using-api.md:340]: Should this be here? We don't have a library for Rust, and this looks like it's an AssemblyScript runnable?

**Review by ospencer — CHANGES_REQUESTED** [review]

File: .github/workflows/documentation.yml
Comment (LauraLangdon) [reply .github/workflows/documentation.yml:16]: IIRC`16.14` was specified when npm was yelling at me. I asked myself the same question at the time.

File: website/docs/compute/compute.md
Comment (LauraLangdon) [reply website/docs/compute/compute.md:6]: Will do next time!

File: website/docs/compute/customizing-functions/code-editor.md
Comment (LauraLangdon) [reply website/docs/compute/customizing-functions/code-editor.md:23]: I think this is what prompted Oscar to file https://github.com/suborbital/docs/issues/162. I can fix this one manually in the meantime.

File: website/docs/compute/customizing-functions/fully-qualified-function-names.md
Comment (LauraLangdon) [reply website/docs/compute/customizing-functions/fully-qualified-function-names.md:19]: Nope! Will fix.

File: website/docs/compute/customizing-functions/fully-qualified-function-names.md
Comment (LauraLangdon) [reply website/docs/compute/customizing-functions/fully-qualified-function-names.md:37]: Yep!

File: website/docs/compute/get-started.md
Comment (LauraLangdon) [reply website/docs/compute/get-started.md:81]: Now tracked here: https://github.com/suborbital/docs/issues/163

File: website/docs/compute/get-started.md
Comment (LauraLangdon) [reply website/docs/compute/get-started.md:86]: Sure does!

File: .github/workflows/documentation.yml
Comment (LauraLangdon) [reply .github/workflows/documentation.yml:18]: I'll remove it!

File: .github/workflows/documentation.yml
Comment (LauraLangdon) [reply .github/workflows/documentation.yml:16]: I just tried changing it to`16` and nothing seemed to break, so all good.

File: website/docs/compute/using-api.md
Comment (LauraLangdon) [reply website/docs/compute/using-api.md]: I'll change the language to `ts` and delete the comment.

Comment (LauraLangdon) [reply website/docs/compute/using-api.md]: Cheers! See previous resolution.

File: website/docs/compute/using-api.md
Comment (ospencer) [inline website/docs/compute/using-api.md:333]:
```suggestion
    let inStr = String.UTF8.decode(input)
```

File: website/docs/compute/using-api.md
Comment (ospencer) [inline website/docs/compute/using-api.md:335]:
```suggestion
    let out = "Hello there, " + inStr
```

File: website/docs/compute/using-api.md
Comment (ospencer) [inline website/docs/compute/using-api.md:337]:
```suggestion
    return String.UTF8.encode(out)
```

File: website/docs/compute/get-started.md
Comment (LauraLangdon) [reply website/docs/compute/get-started.md:78]: Here's the issue! https://github.com/suborbital/docs/issues/160

File: website/docs/compute/get-started.md
Comment (LauraLangdon) [reply website/docs/compute/get-started.md:81]: Here's the issue! https://github.com/suborbital/docs/issues/164

File: website/.idea/.gitignore
Comment (LauraLangdon) [reply website/.idea/.gitignore]: Fixed!

**Comment by LauraLangdon** [comment]
Re: https://github.com/suborbital/docs/pull/155#pullrequestreview-1074226752 from @callahad:

> The `website/docs/assets/editor-screen.png` image still seems _huge_ (over 300k!)... I think we should crop it further, reduce its resolution, and send it through a compressor.

I'm concerned that if it's smaller or lower res it'll be impossible to read. If it doesn't need to be easy to read, I could just remove it instead! I'll create an issue to track this so we can discuss it when you get back: https://github.com/suborbital/docs/issues/165
 
> * It seems like many of the prose changes switch from one sentence per line to hard-wrapping at ~80 characters. Others do not hard wrap. Others seem to break lines at arbitrary locations. Do we care if this is consistent?

I think this will be addressed by https://github.com/suborbital/docs/issues/162, but I'll also spend some more time trying to find that setting in my IDE.

**Comment by flaki** [comment]
@callahad @LauraLangdon I've re-created the screenshot and re-compressed it with OxiPNG (using [Squoosh](https://squoosh.app)):
functioneditor-oxi

It should be roughly ~66KB now, it is a 934x525px@2x image (for retina screens), 934px is the maximum extent of the content column of the docs. Part of the reason the original was so large it seems is that it was converted from a lossy JPEG to a lossless PNG so the artifacts were giving a hard time to the compressor.

**Comment by callahad** [comment]
Here it is at 28 KB:

editor-screenshot

**Comment by LauraLangdon** [comment]
> @callahad @LauraLangdon I've re-created the screenshot and re-compressed it with OxiPNG (using [Squoosh](https://squoosh.app)): functioneditor-oxi
> 
> It should be roughly ~66KB now, it is a 934x525px@2x image (for retina screens), 934px is the maximum extent of the content column of the docs. Part of the reason the original was so large it seems is that it was converted from a lossy JPEG to a lossless PNG so the artifacts were giving a hard time to the compressor.

Thanks, Flaki!

**Comment by LauraLangdon** [comment]
> Here it is at 28 KB:
> 
> editor-screenshot

Thanks, Dan!

**Review by ospencer — APPROVED** [review]

**Pull request merged**
